app/src/main/kotlin/com/simplemobiletools/clock/services/timerService.kt

```diff
@@ -0,0 +1,158 @@
+package com.simplemobiletools.clock.services
+
+import android.annotation.TargetApi
+import android.app.Notification
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.Service
+import android.content.Context
+import android.content.Intent
+import android.os.Build
+import android.os.CountDownTimer
+import android.os.IBinder
+import androidx.core.app.NotificationCompat
+import com.simplemobiletools.clock.R
+import com.simplemobiletools.clock.extensions.config
+import com.simplemobiletools.clock.extensions.getOpenTimerTabIntent
+import com.simplemobiletools.clock.extensions.getTimerNotification
+import com.simplemobiletools.clock.extensions.secondsToMillis
+import com.simplemobiletools.clock.helpers.TIMER_NOTIF_ID
+import com.simplemobiletools.clock.helpers.TIMER_RUNNING_NOTIF_ID
+import com.simplemobiletools.commons.extensions.getFormattedDuration
+import com.simplemobiletools.commons.helpers.isOreoPlus
+import org.greenrobot.eventbus.EventBus
+import org.greenrobot.eventbus.Subscribe
+import org.greenrobot.eventbus.ThreadMode
+
+fun Context.startTimerService() {
```

Comment:
move it into the context extensions to have them in one place

Reply:
this is not generally extension. It is not applicable again to other conditions/services and therefore I would not move it. This function is directly related to TimerService and therefore it is convenient when everything is in one place.

Reply:
if it is related to this only, make it a casual function. Extensions are not created for these cases.